lib/srv/monitor.go

```diff
@@ -141,7 +144,16 @@ func StartMonitor(cfg MonitorConfig) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	go w.start(lockWatch)
+	go func() {
+		w.start(lockWatch)
+		if w.MonitorCloseChannel != nil {
+			// Non blocking send to the close channel.
```

Comment:
Wouldn't the close make this unusable by other monitors? For example, say I'm trying to test a relatively complex bit of logic and multiple monitors could fire -- it's possible that I'm waiting for multiple signals on this channel.

Reply:
Yeah, I suppose if you call `StartMonitor` multiple times with the same channel it would not work. But is that a use case we actually have to support?

Maybe moving the channel out of the config and as a second argument to `StartMonitor` would make that less likely to happen (and we could just pass an explicit nil if we don't care about close notifications)

Reply:
(There's prior art here, BTW. `context.Done()` gives you a channel that is closed when the context is closed, not a channel that is written to)

Reply:
Understood.